tests/spec/llm_spec.lua

```
@@ -0,0 +1,340 @@
+-- Tests for caramba.llm module
+-- Comprehensive test suite covering LLM providers, API calls, and error handling
+
+-- Mock vim API for testing
+local mock_vim = {
+  fn = {
+    jobstart = function(cmd, opts)
+      -- Mock successful job
+      if opts and opts.on_exit then
+        vim.schedule(function()
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'vim'

(W113)